### ISSUE TITLE
Rename the `weave`ed file?

### ISSUE THREAD

**Issue by likanzhan** [JunoLab/Weave.jl#383]
Is it possible to rename the weaved file when using the `weave` function?
Suppose my original file is a plain `index.md` file,  and I want to render it into a `ZZZ.tex` file, using `weave`.
I've tried to achieved this via the `out_path` option, i.e., `out_path = `ZZZ.tex``, but it doesn't work.
Any suggestions?

**Comment by aviatesk** [comment]
Yeah, I found `out_path` option is currently just specifying the output directory; we need to modify:

https://github.com/JunoLab/Weave.jl/blob/b5ba227e757520f389a6d6e0f2cacb731eab8b12/src/Weave.jl#L246-L247

**Comment by pgunnink** [comment]
This bug has broken the Weave implementation in the VS Code Julia extension (see https://github.com/julia-vscode/julia-vscode/issues/1683#issuecomment-788898402). I had a quick look in the code, and it would be a simple case of modifying https://github.com/JunoLab/Weave.jl/blob/b5ba227e757520f389a6d6e0f2cacb731eab8b12/src/Weave.jl#L247 to give the proper `out_path`.

But I am not sure if this breaks existing behavior that relies on this bug. Do you guys want me to create a PR?

**Comment by pfitzseb** [comment]
Yes please. Worst case we could release this as a new major version, but I don't think that'll be necessary.
